### PR TITLE
Socint 36 junit 5

### PR DESCRIPTION
**Pull request by rpcatling**
Junit 4 --> Junit 5.

JIRA LINK: https://collaborate2.ons.gov.uk/jira/browse/SOCINT-36

### CODE REVIEW

File: src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImplGetUACTest.java [hunk 1]
```diff
@@ -69,38 +69,36 @@ public void testGetUACSPGCaseForIndividual() throws Exception {
...
   @Test
   public void testGetUACHICase() {
-    try {
-      doGetUACTest("HI", false);
-      fail();
-    } catch (Exception e) {
-      assertTrue(e.getMessage(), e.getMessage().contains("must be SPG, CE or HH"));
-    }
+    Exception e = assertThrows(Exception.class, () -> doGetUACTest("HI", false));
+    assertTrue(e.getMessage().contains("must be SPG, CE or HH"));
```
Comment (PeterBochel) [inline src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImplGetUACTest.java:73]: Another one.

File: src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImplGetUACTest.java [hunk 2]
```diff
@@ -156,13 +154,9 @@ public void shouldRejectCeManagerFormFromEstabRegionN() {
...
-    try {
-      doGetUACTest(false, FormType.C);
-      fail();
-    } catch (CTPException e) {
-      assertEquals(Fault.BAD_REQUEST, e.getFault());
-      assertTrue(e.getMessage(), e.getMessage().contains(expectedMsg));
-    }
+    CTPException e = assertThrows(CTPException.class, () -> doGetUACTest(false, FormType.C));
+    assertEquals(Fault.BAD_REQUEST, e.getFault());
+    assertTrue(e.getMessage().contains(expectedMsg));
```
Comment (PeterBochel) [inline src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImplGetUACTest.java:159]: Another

File: src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImplLaunchTest.java [hunk 3]
```diff
@@ -171,50 +171,47 @@ public void shouldLaunchWelshCaseWithAddressLevel_E_forCE() throws Exception {
...
   @Test
   public void testLaunchHICase() {
-    try {
-      doLaunchTest("HI", false);
-      fail();
-    } catch (Exception e) {
-      assertTrue(e.getMessage(), e.getMessage().contains("must be SPG, CE or HH"));
-    }
+    Exception e = assertThrows(Exception.class, () -> doLaunchTest("HI", false));
+    assertTrue(e.getMessage().contains("must be SPG, CE or HH"));
```
Comment (PeterBochel) [inline src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImplLaunchTest.java:175]: Another

File: src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImplLaunchTest.java [hunk 4]
```diff
@@ -171,50 +171,47 @@ public void shouldLaunchWelshCaseWithAddressLevel_E_forCE() throws Exception {
...
-    try {
-      doLaunchTest(false, dto, FormType.C);
-      fail();
-    } catch (CTPException e) {
-      assertEquals(expectedFault, e.getFault());
-      assertTrue(e.getMessage(), e.getMessage().contains(expectedMsg));
-    }
+    CTPException e = assertThrows(CTPException.class, () -> doLaunchTest(false, dto, FormType.C));
+    assertEquals(expectedFault, e.getFault());
+    assertTrue(e.getMessage().contains(expectedMsg));
```
Comment (PeterBochel) [inline src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImplLaunchTest.java:214]: Another

**Review by PeterBochel — CHANGES_REQUESTED** [review]
Minor comment about lack of debugging information for assertTrue() failures.

File: src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImplLaunchTest.java (hunk 4 above)
Comment (rpcatling) [reply src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImplLaunchTest.java]: done

File: src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImplLaunchTest.java (hunk 3 above)
Comment (rpcatling) [reply src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImplLaunchTest.java]: done

File: src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImplGetUACTest.java (hunk 2 above)
Comment (rpcatling) [reply src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImplGetUACTest.java]: done

File: src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImplGetUACTest.java (hunk 1 above)
Comment (rpcatling) [reply src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImplGetUACTest.java]: done

File: src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImplCreateCaseForNewAddressTest.java [hunk 5]
```diff
@@ -268,7 +266,7 @@ public void testNewCaseForNewAddress_rejectScottishPostcode() throws Exception {
       fail();
     } catch (CTPException e) {
       assertEquals(Fault.BAD_REQUEST, e.getFault());
-      assertTrue(e.toString(), e.getMessage().startsWith("Scottish addresses are not valid"));
+      assertTrue(e.getMessage().startsWith("Scottish addresses are not valid"));
```
[the remark this answers is not in the document]
Comment (rpcatling) [reply src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImplCreateCaseForNewAddressTest.java]: done

File: src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImplCreateCaseForNewAddressTest.java [hunk 6]
```diff
@@ -213,7 +211,7 @@ private void rejectCrownDependencyPostcodes(String postcode, String expectedErro
       fail();
     } catch (CTPException e) {
       assertEquals(Fault.BAD_REQUEST, e.getFault());
-      assertTrue(e.toString(), e.getMessage().matches(expectedErrorMessage));
+      assertTrue(e.getMessage().matches(expectedErrorMessage));
```
[the remark this answers is not in the document]
Comment (rpcatling) [reply src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImplCreateCaseForNewAddressTest.java]: done

File: src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImplCreateCaseForNewAddressTest.java [hunk 7]
```diff
@@ -142,7 +141,7 @@ public void testNewCaseForNewAddress_ceWithNullNumberOfResidents() {
       fail();
     } catch (CTPException e) {
       assertEquals(Fault.BAD_REQUEST, e.getFault());
-      assertTrue(e.toString(), e.getMessage().matches(".*Number of residents .* for CE .*"));
+      assertTrue(e.getMessage().matches(".*Number of residents .* for CE .*"));
```
[the remark this answers is not in the document]
Comment (rpcatling) [reply src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImplCreateCaseForNewAddressTest.java]: done

File: src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImplCreateCaseForNewAddressTest.java [hunk 8]
```diff
@@ -122,7 +121,7 @@ public void testNewCaseForNewAddress_ceWithNonPositiveNumberOfResidents() {
...
```
[the remark this answers is not in the document]
Comment (rpcatling) [reply src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImplCreateCaseForNewAddressTest.java:125]: done

**Review by PeterBochel — APPROVED** [review]

**Pull request merged**